packages/extension-chrome/src/services/ownership/backend.ts

```diff
@@ -31,24 +31,216 @@ const _Secp256k1Blake160ScriptInfoCache = new Map<NetworkId, ScriptTemplate>();
 // TODO implement the backend
 /* istanbul ignore next */
 export function createBackend(_payload: { rpc: string }): Backend {
+  // TODO use rpc to fetch onchain data when lumos rpc is ready to use in chrome extension
   const rpc = new RPC(_payload.rpc);
 
   return {
-    getSecp256k1Blake160ScriptConfig: () => {
-      errors.unimplemented();
+    getSecp256k1Blake160ScriptConfig: async (): Promise<ScriptConfig> => {
+      const lumosConfig = getConfig();
+      let secp256k1Config = lumosConfig.SCRIPTS['SECP256K1_BLAKE160'];
+      if (!secp256k1Config) {
+        secp256k1Config = await loadSecp256k1ScriptDep(rpc);
+      }
+      return Promise.resolve(secp256k1Config);
     },
-    hasHistories: () => {
-      errors.unimplemented();
+    hasHistories: async (payload: { locks: Script[] }): Promise<boolean[]> => {
+      const toRequestParam = (lock: Script, id: number) => ({
+        id,
+        jsonrpc: '2.0',
+        method: 'get_transactions',
+        params: [
+          {
+            script: {
+              code_hash: lock.codeHash,
+              hash_type: lock.hashType,
+              args: lock.args,
+            },
+            script_type: 'lock',
+          },
+          'asc',
+          '0x1',
+        ],
+      });
+      const requestParam = payload.locks.map(toRequestParam);
+      const rawResponse = await fetch(_payload.rpc, {
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(requestParam),
+        method: 'POST',
+      });
+      const responses = await rawResponse.json();
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const result = responses.map((response: any) => response.result.objects.length > 0);
+      return result;
     },
-    getLiveCellsByLocks: () => {
-      errors.unimplemented();
+    getLiveCellsByLocks: async ({
+      locks,
+      cursor,
+      limit = 20,
+    }: {
+      locks: Script[];
+      /**
+       * the cursor is the ckb indexer cursor
+       */
+      cursor?: string;
+      // TODO implement limit
+      limit?: number;
+    }): Promise<Paginate<Cell>> => {
+      const toRequestParam = (lock: Script, id: number, cursor?: string) => ({
+        id,
+        jsonrpc: '2.0',
+        method: 'get_cells',
+        params: [
+          {
+            script: {
+              code_hash: lock.codeHash,
+              hash_type: lock.hashType,
+              args: lock.args,
+            },
+            script_type: 'lock',
```

Comment:
```suggestion
            script_type: 'lock',
            script_search_mode: 'exact',
```

the indexer defaults to prefix match, we should use the [exact](https://github.com/nervosnetwork/ckb-indexer#get_cells) match here to avoid unexpected cells from the indexer

Reply:
updated in `get_transactions` too